### ISSUE TITLE
issue with transfer-encoding. 

### ISSUE THREAD

**Issue by martingg88** [balderdashy/sails#2980]
i do have issues for following code? any idea. 

1. transfer-coding is not set in response header.

2. res.write doesn't output in browser instantly. but it will only output once the  res.end is hit. 

 function(req, res){

        res.writeHead(200, {'Content-Type':'text/plain;charset=UTF-8', 'Transfer-Encoding': 'chunked'});
        res.write("testing\r\n");
        res.write("testing\r\n");
        res.write("testing\r\n");

        setTimeout(function(){
            return res.end('end');
        }, 60000)

    }

**Comment by nikhiljohn10** [comment]
I have tested this and found working. I couldn't reproduce this. Here is the screen shot of the test.

![screenshot from 2015-06-02 15 19 45](https://cloud.githubusercontent.com/assets/5616249/7933196/1e9fefcc-093b-11e5-8847-5d0fb5aa4bf2.png)

**Comment by martingg88** [comment]
lets forget about transfer-coding maybe there is a proxy at my side. 

do your server send output to browser instantly or wait after timeout?

**Comment by nikhiljohn10** [comment]
the "testing" is shown in browser as the page loads. But "end" is shown only after the timeout.

**Comment by nikhiljohn10** [comment]
Try this. Not sure. but might help you.

```
        res.writeHead(200, {
            'Content-Type': 'text/plain;charset=UTF-8',
            'Transfer-Encoding': 'chunked'
        });
        res.write("testing1\r\n");
        res.write("testing2\r\n");
        res.write("testing3\r\n",function() {
            return res.end('end');
        });
```

**Comment by martingg88** [comment]
i'm trying to find out the issue why it doesn't display "testing" instantly, but display all string at once after timeout. I need to write the progress status to browser while uploading large file to server.

**Comment by nikhiljohn10** [comment]
Have you checked SO??

http://stackoverflow.com/questions/26206068/handling-uploads-with-skipper-in-sails-js-on-progress

**Comment by martingg88** [comment]
sorry i wrongly explained the last post. 

Act i do have script that will be activated by browser and download large file from other server. In between, i expect to send progress status to browser while downloding. i think the script i posted in this discussion should be able to simulate to this workflow. Do you have any idea how come it only send the all of the output (etc: 'testing') to browser after timeout? I thought it should send the output for each res.write instantly. isn't it?

**Comment by nikhiljohn10** [comment]
Yes it will. Try that code is a fresh sails app.

**Comment by martingg88** [comment]
thanks. keep me posted thanks.

**Comment by martingg88** [comment]
tried it in fresh app and still have the same issue.

**Comment by nikhiljohn10** [comment]
Maybe some issue with some module version. Maybe node or http or express. Not sure. Maybe try it on some other system with freshly installed node & sails if you have. I'm using node 0.10.36 & sails 0.11.0. Its working good. If you find the issue persist with same version in your system, then it not a issue with sails.

**Comment by sailsbot** [comment]
Thanks for posting, @martingg88.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
